index.js

```diff
@@ -127,22 +133,27 @@ function main(argv, onResponse) {
 function tcurl(opts) {
     var logger = Logger(opts);
     var spec;
+
     if (opts.thrift) {
-        var specs = {};
-        var files = fs.readdirSync(opts.thrift);
-        files.forEach(function eachFile(file) {
-            var match = /([^\/]+)\.thrift$/.exec(file);
-            if (match) {
-                var serviceName = match[1];
-                var fileName = match[0];
-                specs[serviceName] =
-                    fs.readFileSync(path.join(opts.thrift, fileName), 'utf8');
-            }
-        });
+        var optStat = fs.statSync(opts.thrift);
```

Comment:
Best practice to avoid race conditions: just try to read as file, then if that errs, try to read as directory.